src/azul/queues.py

```diff
@@ -148,6 +149,56 @@ def _is_azul_queue(self, queue_name) -> bool:
         return any(config.unqualified_resource_name_or_none(queue_name, suffix)[1] == config.deployment_stage
                    for suffix in ('', '.fifo'))
 
+    @classmethod
+    def count_messages(cls, queues: Mapping[str, Any]) -> int:
+        """
+        Count the number of messages in the given queues
+
+        :param queues: A collection of mapped queue_names to queues
+        :return: Total count of available, in flight, and delayed messages
+        """
+        attribute_names = ['ApproximateNumberOfMessages' + suffix for suffix in ('', 'NotVisible', 'Delayed')]
+        total_message_count = 0
+        for queue_name, queue in queues.items():
+            queue.reload()
+            message_counts = [int(queue.attributes[attribute_name]) for attribute_name in attribute_names]
+            queue_length = sum(message_counts)
+            logger.debug('Queue %s has %i message(s) (%i available, %i in flight and %i delayed).',
+                         queue_name, queue_length, *message_counts)
+            total_message_count += queue_length
+        logger.info('Counting %i message(s) in %i queue(s).', total_message_count, len(queues))
+        return total_message_count
+
+    @classmethod
+    def wait_for_queue_level(cls, queue_names, empty: bool = True, timeout: int = 60):
+        """
+        Wait until the total count of queued messages reaches the desired level
+
+        :param queue_names: The names of the queues to check messages counts in
+        :param empty: If we should wait until the queues are empty. False to wait until not empty.
+        :param timeout: Give up waiting after this many seconds
+        """
+        queues = {queue_name: boto3.resource('sqs').get_queue_by_name(QueueName=queue_name)
+                  for queue_name in queue_names}
```

Comment:
… and this section could be consolidated. I have the feeling that you will end up with the added Queue methods being instance methods instead of class methods. You *may* have to tackle #1448 as well but I'm not sure.

Reply:
While it wasn't necessary to resolve #1448 here I felt it was related to the work in the previous commit and easy enough so I committed a fix for it in this branch as a separate commit.